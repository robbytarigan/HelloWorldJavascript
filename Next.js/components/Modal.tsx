import React from 'react';
import ReactModal from 'react-modal';


type State = {
    showModal : boolean;
}

class Modal extends React.Component<{}, State> {
  constructor (props : {}) {
    super(props);

ReactModal.setAppElement('#main');
    this.state = {
      showModal: false
    };
    
    this.handleOpenModal = this.handleOpenModal.bind(this);
    this.handleCloseModal = this.handleCloseModal.bind(this);
  }
  
  handleOpenModal () {
    this.setState({ showModal: true });
  }
  
  handleCloseModal() {
    this.setState({ showModal: false });
  }
  
  render =() => (
      <div>
        <button onClick={this.handleOpenModal}>Trigger Modal</button>
        <ReactModal 
           isOpen={this.state.showModal}
           contentLabel="Minimal Modal Example"
        >
          <button onClick={this.handleCloseModal}>Close Modal</button>
        </ReactModal>
      </div>
    );
  
}

export default Modal;