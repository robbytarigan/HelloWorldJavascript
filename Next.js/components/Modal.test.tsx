import React from 'react';
import {shallow} from 'enzyme';
import ReactModal from 'react-modal';
import Modal from './Modal';
jest.mock('react-modal');

// let ReactModal = require('react-modal');
// const {setAppElement} = jest.requireActual('react-modal');
// (ReactModal.setAppElement as jest.Mock).mockImplementation(() => jest.fn());

test('Div renders', () => {
    expect(shallow(<div>test</div>)).toMatchSnapshot();
});

test('Modal renders', () => {
    expect(shallow(<Modal />)).toMatchSnapshot();
    expect((ReactModal.setAppElement as jest.Mock).mock.calls.length).toBe(1);
});