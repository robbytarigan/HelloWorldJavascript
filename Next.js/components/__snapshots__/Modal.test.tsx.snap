// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Div renders 1`] = `
<div>
  test
</div>
`;

exports[`Modal renders 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Trigger Modal
  </button>
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    closeTimeoutMS={0}
    contentLabel="Minimal Modal Example"
    isOpen={false}
    parentSelector={[MockFunction]}
    portalClassName="ReactModalPortal"
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <button
      onClick={[Function]}
    >
      Close Modal
    </button>
  </Modal>
</div>
`;
